### ISSUE TITLE
Type 'Foo' does not conform to protocol '_MessageImplementationBase'

### ISSUE THREAD

**Issue by AaronTriplett** [apple/swift-protobuf#790]
I am generating my pb.swift files outside of my project using the Protoc compiler and protoc_gen_swift + protoc-gen-swiftgrpc plugins. 

The extensions for each message type in my generated pb.swift files do not conform to the _MessageImplementationBase protocol. Looks like they are missing the _protobuf_generated_isEqualTo function.

<img width="869" alt="screen shot 2018-08-31 at 11 52 20 pm" src="https://user-images.githubusercontent.com/31156329/44943454-63198680-ad7b-11e8-93a3-4d7af541d791.png">

Name | Value
------------ | -------------
OS | **macOS**
Xcode | **9.4.1**
Swift | **4.1**
Protoc | **3.5.1**
Swift-Protobuf | **1.1.1**
gRPC-Swift | **0.5.1**

**Comment by AaronTriplett** [comment]
After some more digging, it looks like this was changed as of this commit: (https://github.com/apple/swift-protobuf/commit/56aa257787c9ed07566efbec68393e5d5531ec4e) in the 1.1.1 release (401e774). **Everything works fine in 1.1.0.**

This results in a compilation error and seems like a pretty big issue to me unless I'm missing something.

**Comment by AaronTriplett** [comment]
It looks like I was missing something. 😁

I'm building the swift + swiftgrpc plugins on master from **grpc-swift** which in turn has a dependency on **swift-protobuf** 1.1.1. Since I'm generating my pb.swift files using the plugins from **grpc-swift** master, my message types do not have the _protobuf_generated_isEqualTo function (since it now has a default implementation under the _MessageImplementationBase extension). 

However, when including SwiftGRPC in my podspec, it's grabbing their latest 0.5.1 release which uses **swift-protobuf** 1.0.3 and in-turn the default implementation extension for _MessageImplementationBase is not included.

Updating to target their 0.5.1 release fixes my problem, and I'll be closing this out.

**Comment by blackivory86** [comment]
TLDR:
Update the swift-protobuf library of your Xcode-project to a newer version that matches the protobuf generator

**Comment by AaronTriplett** [comment]
Or, the alternative, use the official releases for the plugins and project libraries. This was a breaking API change, didn't expect it from a minor release (1.1.0 -> 1.1.1)...

**Comment by thomasvl** [comment]
@AaronTriplett - I believe your issue was from using a newer generator with an older library.  I believe that change was done so the new library supported old generated code; but the direction you were using is just about impossible without forcing you to specific the version you want to generate for (instead of assuming it is for the version of the generator being used).  The other choice is to not bother with the support we did add and just always major version bump on any change, but that can end up also being somewhat unfriendly also.  In anycase, we likely should have messaged it better in the release notes.  😦

**Comment by blackivory86** [comment]
@BObereder 
Either use an older generator to match your swift-protobuf 1.2 lib or update the lib to the latest 1.3.1

**Comment by BObereder** [comment]
@blackivory86 homebrew sais swift-protobuf is up to date. Any idea why it doesnt allow me to install the newest version?

**Comment by blackivory86** [comment]
@BObereder 
The problem is probably an old swift-protobuf lib in your Xcode-Project. Update that one through Carthage/Cocoapods/whatever

**Comment by BObereder** [comment]
@blackivory86 you are right that was the problem 🎉
thanks for the hint in the right direction.
